packages/flutter/lib/src/material/ink_ripple.dart

```diff
@@ -199,9 +199,14 @@ class InkRipple extends InteractiveInkFeature {
   @override
   void cancel() {
     _fadeInController.stop();
-    _fadeOutController
-      ..value = 1.0 - _fadeInController.value
-      ..animateTo(1.0, duration: _kCancelDuration);
+    // Watch out: setting _fadeOutController's value to 1.0 would
+    // trigger a call to _handleAlphaStatusChanged() which would
+    // dispose _fadeOutController.
+    if (_fadeInController.value > 0.0) {
+      _fadeOutController
+        ..value = 1.0 - _fadeInController.value
```

Comment:
for improved readability you may wish to factor out `1.0 - _fadeInController.value` and then in the if statement test that the factored out value is `< 1.0`, then the comment would be clearer